### DISCUSSION TITLE
Congrats to authors!

### DISCUSSION THREAD

**Discussion by jcomand** [jasonacox/Powerwall-Dashboard#291] [General]
This project rocks!  The dashboard is beautiful, the data is amazing, and I can't believe the install with setup.sh worked first time for such a complicated project (Docker via WSL on Windows), with just a couple minutes of setup.  Strong work!!

**Comment by jasonacox** [comment]
Thanks for the kind words an encouragement @jcomand !! 🙏 Made my day! 🤗 

We put these notes together for Win11: https://github.com/jasonacox/Powerwall-Dashboard#windows-11-instructions  - Did you need to use any of that? Any suggestions to help other WSL users?

**Reply by jcomand** [reply]
Glad to hear it!  Sure, happy to provide some tips to help people get started.  Yes, I pretty much followed the instructions, in my case for Windows 10 Home.  I already had docker desktop and git bash and python installed from other projects, so I was pleasantly surprised when I downloaded and ran the setup.sh and it just worked!  (The Grafana instructions are good except it could use one line that says when importing the dashboard, select "influxdb" for the influxdb database and "sun and moon" for the sun and moon data base; it is obvious after you get an error but would be nice.)  But actually the hard part for me was getting the information ready for the setup.sh questions:  

I could connect to the Backup Gateway 2 via ethernet - I did not need to deal with the wifi network the Gateway broadcasts, as you will see in the Tesla instructions.  My Gateway was plugged into my nearby router by the installers- not sure if that makes it a special case.  To start, I had installed your pypowerwall package separately (actually I had found it first).  The scan feature was not working (just hung until I pressed control C, and then showed partial output- maybe I did not wait long enough?), so I got the IP address from my router status screen.  Then I tried various passwords running pypowerwall from the terminal window until I found the correct one.  pypowerwall said could not connect sometimes, and sometimes incorrect password, but eventually I found the correct password and saw the json data displayed.  Again the various generations of online Tesla docs are confusing here.  The password is the last 5 characters of the "password" on the sticker in the vertically oriented Backup Gateway 2 in the basement, NOT on any stickers inside the inverters in the horizontally oriented boxes above the powerwalls, and NOT related to any information available in the app.  Some docs say the password sticker is on the user manual- mine was not.  Nor is it related to the serial number in the app.  I read elsewhere that the email address is essentially ignored, which is good because did not know which email had been set up with my powerwall. 

Once pypowerwall was working, I got my time zone from the output. (Also it looks like you can find the TZ identifiers in the second column of https://en.wikipedia.org/wiki/List_of_tz_database_time_zones ; e.g. America/New_York . Consider adding to the docs. ) Latitude and Longitude I used google maps instead of the provided url in case the location of my browser/ISP was not accurate enough (which is probably silly - anywhere in town would have essentially the same sunrise/sunset/weather).  I provided my tesla app email but I do not think it matters what you give it.  

And it works! The main display is so great, instead of flicking through several graphs on the app.  I was also particularly interested in the string data, because I wanted to see what the output looks like when the system is new.  If there is any damage or decrease in output, I wanted to know how it was when new.  I noted that the voltages were proportional to the number of panels in each string (which the installer told me the panel #s) - turns out it is about 35 volts per panel when the sun is shining.  The amps & power varied with the sun, etc.

I have not checked to see how robust the Windows Docker Desktop setup is to rebooting, crashing etc.  I am not sure if any data is stored inside the Docker container, or if it is all in the user directory.  Sometimes I have deleted containers by accident.  I paused the container before rebooting, but did not test if this is needed.  After reboot, I had to restart my Docker Desktop app (have not made it auto-run yet), but when I did, the container started running again and the old data was obviously still there. 

In the end I was looking for an excuse to run something on a Raspberry Pi, so I will admit I just set it up all over again on that platform!  I know you have backup/restore instructions but I just copied the settings from pypowerwall.env (and the weather API key from the website) and started fresh since I did not want to copy anything cross-platform.

Hope this helps someone.

I really appreciate how beautifully implemented this software and the setup is. I am thinking about what other kind of projects I could use such a dashboard for.

**Reply by jasonacox** [reply]
This is great @jcomand !  A lot of good enhancement suggestions to unpack here.

I made one change related to the TZ documentation: https://github.com/jasonacox/Powerwall-Dashboard/commit/58b8bc1a44c3e51b5019ec816b9e48599e814e48

> To start, I had installed your pypowerwall package separately (actually I had found it first). The scan feature was not working (just hung until I pressed control C,

I would love to figure this out. I suspect it could look like malware to some AV systems (it is doing a sequential IP scan) and be blocked, but I haven't seen that before even on my Win11 setup.  Is this what you ran?

```bash
python3 -m pypowerwall scan
```

**Reply by jcomand** [reply]
Thanks. Yes I used that command for pypowerwall.  I checked again - I was just being impatient.  The pypowerwall scan eventually finishes correctly, but the "running scan..." notification and the rest of the results do not come up until it is finished everything.  So it looks frozen. Same with or without antivirus on.
